syntax = "proto3";

option java_multiple_files = true;
option java_package = "io.pravega.example.simple_grpc_server";
option java_outer_classname = "SimpleGrpcServerProto";
option objc_class_prefix = "PSGS";

package pravega_simple_grpc_server;

service SimpleGrpcServer {
  rpc Test1 (Test1Request) returns (Test1Reply) {}
  rpc Test2 (Test1Request) returns (Test1Reply) {}
  rpc Test3 (Test1Request) returns (stream Test1Reply) {}
  rpc Test4 (Test1Request) returns (stream Test1Reply) {}
  rpc ReadEvents (ReadEventsRequest) returns (stream ReadEventsResponse) {}
  rpc WriteEvents (stream WriteEventsRequest) returns (WriteEventsResponse) {}
}

message Test1Request {
  string name = 1;
}

message Test1Reply {
  string message = 1;
}

message ReadEventsRequest {
  string scope = 1;
  string stream = 2;
}

message ReadEventsResponse {
  bytes event = 1;
  string position = 2;
  string event_pointer = 3;
  // Empty string if this is not a checkpoint.
  string checkpoint_name = 4;
}

message WriteEventsRequest {
  bytes event = 1;
  string scope = 2;
  string stream = 3;
  string routing_key = 4;
}

message WriteEventsResponse {
}
